src/hooks/core/onesignal/useOneSignalSubscribeToCommunityHooks.ts

```diff
@@ -4,15 +4,52 @@ import {useNavigation} from '@react-navigation/core';
 
 import OneSignalUtil from '../../../service/onesignal';
 import useUserAuthHook from '../auth/useUserAuthHook';
+import AnalyticsEventTracking, {
+  BetterSocialEventTracking
+} from '../../../libraries/analytics/analyticsEventTracking';
 import getFeatureLoggerInstance, {EFeatureLogFlag} from '../../../utils/log/FeatureLog';
-import {OneSignalTopicNotificationData} from './types.d';
+import {ONESIGNAL_PN_TYPE} from '../../../utils/constants';
+import {OneSignalNotificationData} from './types.d';
 
 const {featLog} = getFeatureLoggerInstance(EFeatureLogFlag.useOneSignalSubscribeToCommunityHooks);
 
+export type HomeTabsScreen = 'SignedChannelList' | 'AnonymousChannelList' | 'Feed' | 'Profile';
+export type ScreenData = {
+  screen: string;
+  params?: any;
+};
+
 const useOneSignalSubscribeToCommunityHooks = () => {
   const navigation = useNavigation();
   const {token, signedProfileId} = useUserAuthHook();
 
+  const baseResetNavigation = (tabsScreenName: HomeTabsScreen, screen: ScreenData) => {
+    setTimeout(() => {
+      const routes = [
+        {
+          name: 'AuthenticatedStack',
+          params: {
+            screen: 'HomeTabs',
+            params: {
+              screen: tabsScreenName
+            }
+          }
+        },
+        {
+          name: 'AuthenticatedStack',
+          params: {
+            ...screen
+          }
+        }
+      ];
+
+      navigation.reset({
+        index: screen?.screen ? 2 : 1,
+        routes
+      });
+    }, 500);
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `baseResetNavigation` function is using a `setTimeout` with a hardcoded delay of 500ms. This could potentially lead to issues if the operation that needs to be delayed takes longer than expected. It would be better to use Promises or async/await to handle asynchronous operations.

```diff
-    setTimeout(() => {
-      const routes = [
-        {
-          name: 'AuthenticatedStack',
-          params: {
-            screen: 'HomeTabs',
-            params: {
-              screen: tabsScreenName
-            }
-          }
-        },
-        {
-          name: 'AuthenticatedStack',
-          params: {
-            ...screen
-          }
-        }
-      ];
-
-      navigation.reset({
-        index: screen?.screen ? 2 : 1,
-        routes
-      });
-    }, 500);
+    const routes = [
+      {
+        name: 'AuthenticatedStack',
+        params: {
+          screen: 'HomeTabs',
+          params: {
+            screen: tabsScreenName
+          }
+        }
+      },
+      {
+        name: 'AuthenticatedStack',
+        params: {
+          ...screen
+        }
+      }
+    ];
+
+    navigation.reset({
+      index: screen?.screen ? 2 : 1,
+      routes
+    });
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->